### ISSUE TITLE
Support for nested populate?

### ISSUE THREAD

**Issue by DavidBLynch** [florianholzapfel/express-restify-mongoose#137]
Assuming:

```javascript
var postSchema = mongoose.Schema({
    message: String,
    lastModified: {
        user: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
        date: Date
    }
}    
```
then

```
[GET] api.com/posts/?populate=lastModified.user
```

gives this error:

```
TypeError: Cannot read property 'schema' of undefined
    at findExcludedFieldsFor (d:\arp\node_modules\express-restify-mongoose\lib\resource_filter.js:88:50)
    at filterPopulated (d:\arp\node_modules\express-restify-mongoose\lib\resource_filter.js:120:26)
    ...
```

Looking at the code it's trying to break "lastModified.user" into path segments and then search in the schema.paths, but the schema already has the path entry as "lastModified.user" so path segments are not required.

**Comment by Zertz** [comment]
This needs a test to confirm, but ti should work in 2.x

**Comment by nategupt** [comment]
I tried with 2.1.1 but it crashes. 1.x was returning a 400 instead.

/app/node_modules/express-restify-mongoose/lib/resource_filter.js:99
    var path = currSchema.path(fieldArray[fieldArray.length - 1])
                         ^

TypeError: Cannot read property 'path' of undefined
    at findExcludedFieldsFor (/app/node_modules/express-restify-mongoose/lib/resource_filter.js:99:26)
    at filterPopulated (/app/node_modules/express-restify-mongoose/lib/resource_filter.js:140:26)
    at filterObject (/app/node_modules/express-restify-mongoose/lib/resource_filter.js:163:7)
    at EventEmitter.<anonymous> (/app/node_modules/express-restify-mongoose/lib/operations.js:127:23)
    at EventEmitter.<anonymous> (/app/node_modules/mongoose/node_modules/mpromise/lib/promise.js:175:45)
    at emitOne (events.js:77:13)
    at EventEmitter.emit (events.js:169:7)

**Comment by nategupt** [comment]
Actually, my question might be slightly different. I have multiple levels of objects and I am trying to deeply populate them.

var postSchema = mongoose.Schema({
    message: String,
     user: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
    }
} 

var userSchema = mongoose.Schema({
    message: String,
     child: { type: mongoose.Schema.Types.ObjectId, ref: 'Child' },
    }
} 


[GET] api.com/posts/?populate=user.child

**Comment by Zertz** [comment]
@nategupt Indeed, can you create a new issue please?

**Comment by nategupt** [comment]
Filed issue #174

**Comment by Zertz** [comment]
Populating nested paths work, but there is a bug in **resource_filter.js** causing requests to hang. Looking into it.

**Comment by Zertz** [comment]
Fixed in 2.3.0
